jobs/scheduled/ensure_expired_event_destruction.rb

```diff
@@ -28,19 +28,31 @@ def execute(args)
         .where("topics.id IN (?)", calendar_topic_ids)
         .find_each do |pcf|
 
-        details = JSON.parse(pcf.value)
-        details.each do |post_number, (_, from, to, _, recurring)|
+        posts_with_dates = JSON.parse(pcf.value)
+        post_numbers_with_dates = posts_with_dates.keys.map(&:to_i)
+        posts_with_dates.each do |post_number, (_, from, to, _, recurring)|
           next if recurring
 
           to_time = to ? Time.parse(to) : Time.parse(from) + 24.hours
+          next if (to_time + delay.hour) > Time.zone.now
 
-          if (to_time + delay.hour) < Time.zone.now
-            if post = pcf.post.topic.posts.find_by(post_number: post_number)
-              PostDestroyer.new(Discourse.system_user, post).destroy
+          if post = pcf.post.topic.posts.find_by(post_number: post_number)
+            destroy_post(post)
```

Comment:
I'm just wondering, wouldn't be cleaner to try to destroy replies before destroying parent?